Base/Python/slicer/util.py

```diff
@@ -3795,6 +3795,71 @@ def pip_uninstall(requirements):
     _executePythonModule('pip', args)
 
 
+def validate_module_version(
+    module_name: str,
+    minimum_version: str,
+    attempt_fix: bool = False,
+) -> None:
+    """
+    Ensure that some minimum package requirement is met at runtime.
+    
+    If the minimum version requirement is not met, either throw or
+    attempt to install the package and possibly restart Slicer.
+
+    :param module_name: The Python Package Index module to check.
+    :param minimum_version: The minimum version requirement.
+    :param attempt_fix: Whether to attempt to fix an installation or version issue
+        with reinstallation and possible application restart.
+
+    :raises ModuleNotFoundError: if a module satisfying version requirements
+        is not installed and the issue could not be fixed.
+    """
+    import importlib
+    from packaging import version
+
+    needRestart = False
+    needInstall = False
+
+    try:
+        module = importlib.import_module(module_name)
+    except ModuleNotFoundError as e:
+        install_message = f"The current operation requires {module_name}.\nClick OK to install from PyPI."
+        if not attempt_fix or not confirmOkCancelDisplay(
+            install_message
+        ):
+            failure_message = f"{module_name} is not installed"
+            raise e
+        module = None
+        needInstall = True
+
+    if module and version.parse(module.__version__) < version.parse(minimum_version):
+        if not attempt_fix or not confirmOkCancelDisplay(
+            f"The current operation requires {module_name}>={minimum_version}.\n"
+            f"Click OK to upgrade {module_name} and restart the application."
+        ):
+            failure_message = f"Expected {module_name}>={minimum_version} but found {module.__version__}"
+            raise ModuleNotFoundError(failure_message)
+        needRestart = True
+        needInstall = True
+
+    if needInstall:
+        import slicer
+        progressDialog = createProgressDialog(
+            labelText=f"Installing {module_name}. This may take a minute...",
+            maximum=0,
+        )
+        slicer.app.processEvents()
+        try:
+            failure_message = f"Failed to install {module_name}>={minimum_version}."
+            with slicer.util.tryWithErrorDisplay(failure_message):
+                pip_install(f"{module_name}>={minimum_version}")
+        finally:
+            progressDialog.close()
+
+    if needRestart:
+        restart()
```

Comment:
```suggestion
```

I suggest we revisit the generalization of in a another pull request and simply remove the check currently added in `DICOMProcesses`

Instead of `validate_module_version`, having a method called `is_python_module_installed` would be more explicit and focused on doing on task: _Checking that a specific module (and optinally specified version) is installed_

Strictly speaking[^1], raising `ModuleNotFoundError` is not _valid_ in the sense that a module could be installed without having the correct version. I am not sure redefining the meaning of this specific exception in he context of this function is sensible.

Installation, confirmation[^2], progress reporting and restart[^3] should be handled outside of this function.

### Example

An idea would be to support the following:

```
moduleName="dicomweb_client"
versionSpecifier = ">=0.51"

if not slicer.util.is_python_module_installed(
    moduleName, version_specifier=versionSpecifier
):
    if slicer.util.confirmOkCancelDisplay("This current operation requires '{module_name}' Python package. Click OK to install it now."):
        slicer.util.pip_install(f"{moduleName}{versionSpecifier}")
```


[^1]: https://docs.python.org/3/library/exceptions.html#ModuleNotFoundError

[^2]: Handle using the approach suggested in at https://slicer.readthedocs.io/en/latest/developer_guide/script_repository.html#install-a-python-package

[^3]: Restart should also be handled with a confirmation dialog.

Reply:
> I suggest we revisit the generalization of in a another pull request

That's fine. I won't have bandwidth to drive refinement in that separate PR, but if you think this is a worthwhile utility to pursue I can open a draft PR for someone else to take across the finish line.

> and simply remove the check currently added

Will do.

Reply:
Tracked in https://github.com/Slicer/Slicer/pull/7268 and removed from this PR.